PhysicsTools/PatAlgos/plugins/BaseMVAValueMapProducer.h

```diff
@@ -63,19 +73,47 @@ class BaseMVAValueMapProducer : public edm::stream::EDProducer<> {
       values_.resize(variablesOrder_.size());
       size_t i=0;
       for(const auto & v : variablesOrder_){
-	positions_[v]=i;
-	reader_.AddVariable(v,(&values_.front())+i);
-	i++;
+        positions_[v]=i;
+        if(tmva_) reader_->AddVariable(v,(&values_.front())+i);
+	    i++;
       }
 //      reader_.BookMVA(name_,iConfig.getParameter<edm::FileInPath>("weightFile").fullPath() );
-      reco::details::loadTMVAWeights(&reader_, name_, iConfig.getParameter<edm::FileInPath>("weightFile").fullPath());
-      produces<edm::ValueMap<float>>();
+    if(tmva_)
+    {
+        reco::details::loadTMVAWeights(reader_, name_, weightfilename_);
+    }else if(tf_)    {
+        tensorflow::setLogging("3");
+        graph_=tensorflow::loadGraphDef(weightfilename_);
+        inputTensorName_=iConfig.getParameter<std::string>("inputTensorName");
+        outputTensorName_=iConfig.getParameter<std::string>("outputTensorName");
+        output_names_=iConfig.getParameter<std::vector<std::string>>("outputNames");
+        for(const auto & s : iConfig.getParameter<std::vector<std::string>>("outputFormulas")) { output_formulas_.push_back(StringObjectFunction<std::vector<float>>(s));} 
+        size_t nThreads = iConfig.getParameter<unsigned int>("nThreads");
+        std::string singleThreadPool = iConfig.getParameter<std::string>("singleThreadPool");
+        tensorflow::SessionOptions sessionOptions;
+        tensorflow::setThreading(sessionOptions, nThreads, singleThreadPool);
+        session_ = tensorflow::createSession(graph_, sessionOptions);
+
+    } else  {
+        //FIXME: throw config error
```

Comment:
can this be fixed?

Reply:
it could but there was a request to have the code for nanoaod ready in short time.
I suggest to fix this (in the next days) in this PR (10XY) while we proceed as is on the 94X for time reasons.